library/src/main/java/com/novoda/downloadmanager/MigrationExtractor.java

```diff
@@ -56,36 +90,13 @@
         }
     }
 
-    private List<Migration.FileMetadata> extractFileMetadataFrom(String batchId, Batch.Builder newBatchBuilder) {
-        Cursor downloadsCursor = database.rawQuery(DOWNLOADS_QUERY, batchId);
-
-        if (downloadsCursor == null) {
-            return Collections.emptyList();
-        }
-
-        try {
-            List<Migration.FileMetadata> fileMetadataList = new ArrayList<>();
-
-            while (downloadsCursor.moveToNext()) {
-                String originalFileId = downloadsCursor.getString(FILE_ID_COLUMN);
-                String originalNetworkAddress = downloadsCursor.getString(NETWORK_ADDRESS_COLUMN);
-                String originalFileLocation = downloadsCursor.getString(FILE_LOCATION_COLUMN);
-                newBatchBuilder.addFile(originalNetworkAddress).apply();
-
-                FilePath filePath = new LiteFilePath(originalFileLocation);
-                long rawFileSize = filePersistence.getCurrentSize(filePath);
-                FileSize fileSize = new LiteFileSize(rawFileSize, rawFileSize);
-                Migration.FileMetadata fileMetadata = new Migration.FileMetadata(
-                        originalFileId,
-                        originalFileLocation,
-                        fileSize,
-                        originalNetworkAddress
-                );
-                fileMetadataList.add(fileMetadata);
-            }
-            return fileMetadataList;
-        } finally {
-            downloadsCursor.close();
+    private DownloadBatchId createDownloadBatchIdFrom(String originalFileId, String batchId) {
```

Comment:
Client application needs the `DownloadBatchId` to be derived from the first files `notificationExtras` 😬 we will need to provide a mechanism for clients to specify this mapping before a public release.

Reply:
The same applies for `FileIds` too.